crates/next-core/src/next_font_google/mod.rs

```diff
@@ -1,3 +1,272 @@
+    #[turbo_tasks::function]
+    pub fn new(project_path: FileSystemPathVc) -> Self {
+        Self::cell(NextFontGoogleReplacer { project_path })
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl ImportMappingReplacement for NextFontGoogleReplacer {
+    #[turbo_tasks::function]
+    fn replace(&self, _capture: &str) -> ImportMappingVc {
+        ImportMapping::Ignore.into()
+    }
+
+    #[turbo_tasks::function]
+    async fn result(&self, request: RequestVc) -> Result<ImportMapResultVc> {
+        let request = &*request.await?;
+        let Request::Module {
+            module: _,
+            path: _,
+            query,
+        } = request else {
+            return Ok(ImportMapResult::NoEntry.into());
+        };
+
+        let query = &*query.await?;
+        let js_asset = VirtualAssetVc::new(
+                attached_next_js_package_path(self.project_path)
+                    .join("internal/font/google/inter.js"),
+                FileContent::Content(
+                    formatdoc!(
+                        r#"
+                            import cssModule from "@vercel/turbopack-next/internal/font/google/cssmodule.module.css?{}";
+                            export default {{
+                                className: cssModule.className
+                            }};
+                        "#,
+                        // Pass along whichever options we received to the css handler
+                        qstring::QString::new(query.as_ref().unwrap().iter().collect())
+                    )
+                    .into(),
+                )
+                .into(),
+            );
+
+        Ok(ImportMapResult::Result(ResolveResult::Single(js_asset.into(), vec![]).into()).into())
+    }
+}
+
+#[turbo_tasks::value(shared)]
+pub struct NextFontGoogleCssModuleReplacer {
+    project_path: FileSystemPathVc,
+}
+
+#[turbo_tasks::value_impl]
+impl NextFontGoogleCssModuleReplacerVc {
+    #[turbo_tasks::function]
+    pub fn new(project_path: FileSystemPathVc) -> Self {
+        Self::cell(NextFontGoogleCssModuleReplacer { project_path })
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl ImportMappingReplacement for NextFontGoogleCssModuleReplacer {
+    #[turbo_tasks::function]
+    fn replace(&self, _capture: &str) -> ImportMappingVc {
+        ImportMapping::Ignore.into()
+    }
+
+    #[turbo_tasks::function]
+    async fn result(&self, request: RequestVc) -> Result<ImportMapResultVc> {
+        let request = &*request.await?;
+        let Request::Module {
+            module: _,
+            path: _,
+            query,
+        } = request else {
+            return Ok(ImportMapResult::NoEntry.into());
+        };
+
+        let query_map = &*query.await?;
+        // These are invariants from the next/font swc transform. Regular errors instead
+        // of Issues should be okay.
+        let query_map = query_map
+            .as_ref()
+            .context("@next/font/google queries must exist")?;
+
+        assert_eq!(
```

Comment:
This'll segfault the program if the user specified another param to the `NextFont` constructor, right? Can we remove this and give an error instead?